### PR TITLE
reuse the existing deploy script

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Reuse the existing deploy script.

Kubermatic charts need to be installed in a specific order as the chart depend on each other.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by mrIncompetent** [comment]
cc @nikhita

File: api/hack/helper.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
```
Comment (alvaroaleman) [inline api/hack/helper.sh]: Maybe rename to `lib.sh` or something to make clear that this script never gets called by itself? Also wouldn't it make sense to source this script in all other scripts that use the `retry` function?

File: api/hack/ci/ci-deploy-dev-cloud.sh [hunk 2]
```diff
@@ -111,51 +23,46 @@ echodate "Getting secrets from Vault"
...
-  dev/seed-clusters/cloud.kubermatic.io > $VALUES_FILE
+vault kv get -field=kubeconfig dev/seed-clusters/cloud.kubermatic.io > ${KUBECONFIG}
+vault kv get -field=asia-east1-a-1-values.yaml dev/seed-clusters/cloud.kubermatic.io > ${VALUES_FILE}
+kubectl config use-context asia-east1-a-1
 echodate "Successfully got secrets for cloud-asia from Vault"
 
 echodate "Deploying Kubermatic to cloud-asia"
-deploySeed "--kube-context=asia-east1-a-1 --tiller-namespace=kubermatic-installer"
-echodate "Successfully deployed Kubermatic to cloud-asia"
+./api/hack/deploy.sh seed ${VALUES_FILE} ${HELM_EXTRA_ARGS}
```
Comment (alvaroaleman) [inline api/hack/ci/ci-deploy-dev-cloud.sh:50]: Why call out into another script and not have everything in one script?

`deploy.sh` makes it look like it was some generic thing but it really isn't, its just used for dev and cloud

Comment (mrIncompetent) [reply api/hack/ci/ci-deploy-dev-cloud.sh:50]: I used the script to deploy kubermatic in the past.

But as you mentioned it, i'll update the `ci-deploy-run.sh` script as well

File: api/hack/ci/ci-deploy-run.sh [hunk 3]
```diff
@@ -81,18 +23,20 @@ echodate "Getting secrets from Vault"
...
 export VAULT_TOKEN="$(cat /tmp/vault-token-response.json| jq .auth.client_token -r)"
 export KUBECONFIG=/tmp/kubeconfig
 export VALUES_FILE=/tmp/values.yaml
-
-# deploy to drun
-vault kv get -field=kubeconfig \
-  dev/seed-clusters/run.kubermatic.io > $KUBECONFIG
-vault kv get -field=values.yaml \
-  dev/seed-clusters/run.kubermatic.io > $VALUES_FILE
+export HELM_EXTRA_ARGS="--tiller-namespace=kubermatic-installer \
```
Comment (alvaroaleman) [inline api/hack/ci/ci-deploy-run.sh]: The tiller namespace on run is `kube-system`

Comment (mrIncompetent) [reply api/hack/ci/ci-deploy-run.sh]: Fixed in https://github.com/kubermatic/kubermatic/pull/2953/commits/eab068a34edc1c81114fc9a9d380d66b7d370272

File: api/hack/helper.sh (hunk 1 above)
Comment (mrIncompetent) [reply api/hack/helper.sh]: Done in https://github.com/kubermatic/kubermatic/pull/2953/commits/e52c8a4bc142552d0685e4801c9e724e394c767a + https://github.com/kubermatic/kubermatic/pull/2953/commits/afd808e209dfb371d255c65cdbeb3125357ca2da

**Pull request merged**